### PR TITLE
feature: Implement EF entities and configure the DbContext

### PR DESCRIPTION
**Pull request by the-17th-fox**
### What & How

Implement entities according to the db schema. Users and DbContext should support the Identity.
Add shadow properties for the entities (updatedAt, createdAt and IsDeleted for the user). Configure the on delete behaviour for the entities.
Then, override SaveChangesAsync method to work with UpdatedAt and CreatedAt statutes.
After all configure the main class.

### How

Every entity is derived from BasicEntity (consists of an Id field only), as exception User derives also the IdentityUser class.
Lots and auctions use enums with status codes.
In the context were added shadows properties as described above.
Overrided SaveChangesAsync method to work with edited and newly created entries. If an entry modified/created and has such field as UpdatedAt or CreatedAt, SaveChangesMethod should update the data in this properties.
In the main class the DbContext and Identity config were added to the builder.

### Affected Area

CA.WebAPI
Core/CA.Domain.Entities
Core/CA.Domain.Enums
Infrastructure/CA.Persistence.Context
Infrastructure/Migrations

### PR Checklist

*Build*

 - [x] Build Succeeded

### CODE REVIEW

File: Core/CA.Domain/Entities/User.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System.Threading.Tasks;
+
+namespace CA.Domain.Entities
+{
+    public class User : IdentityUser<int>
+    {
+        public string FirstName { get; set; } = String.Empty;
+        public string LastName { get; set; } = String.Empty;
+        public List<Transaction> Transactions { get; set; }
+        public List<LotBid> Bids { get; set; } = new();
```
Comment (TsvirkoNikita) [inline Core/CA.Domain/Entities/User.cs:15]: What is the LotBid? Looks like a typo

File: Core/CA.Domain/Entities/Lot.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    public class Lot : BasicEntity
+    {
+        public int CarId { get; set; }
+        public Car Car { get; set; }
+
+        public int AuctionId { get; set; }
+        public Auction? Auction { get; set; }
+
+        public int LastBidId { get; set; }
+        public LotBid? LastBid { get; set; }
```
Comment (TsvirkoNikita) [inline Core/CA.Domain/Entities/Lot.cs:19]: The same

**Comment by TsvirkoNikita** [comment]
After you will be done with the requested changes, add a migration

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]

File: Core/CA.Domain/Entities/Lot.cs (hunk 2 above)
Comment (the-17th-fox) [reply Core/CA.Domain/Entities/Lot.cs:19]: Thanks, got it

**Comment by the-17th-fox** [comment]
Fixed all bugs (with relationships and small typos)
Added non-nullable properties (as well updated DbSets in the context)
Implemented overrided method for SaveChangesAsync
Configured the DeleteBehavior for some tables
And added new initial migration

**Comment by TsvirkoNikita** [comment]
Please, think about reworking the relationship between the Car and Lot entities to one-to-many. A lot can be canceled and unclaimed, so, cars could have related to multiple lot entities.

**Review by TsvirkoNikita — APPROVED** [review]

**Pull request merged**